test/e2e/generator/generate.go

```diff
@@ -196,6 +204,12 @@ func generateNode(r *rand.Rand, mode e2e.Mode, startAt int64, forceArchive bool)
 		node.SnapshotInterval = 3
 	}
 
+	if node.Mode == "validator" {
+		misbehaviorHeight := startAt + misbehaviorHeighDelta.Choose(r).(int64)
+		node.Misbehaviors = nodeMisbehaviors.Choose(r).(misbehaviorOption).atHeight(misbehaviorHeight)
+		node.PrivvalProtocol = "file"
```

Comment:
This isn't correct, you're now forcing `file` for all validators. You should only force it when any misbehaviors are defined.

Reply:
right good spot